test/e2e/tests/request-queuing/ui.spec.ts

```diff
@@ -32,19 +32,27 @@ const { CaveatTypes } = require('../../../../shared/constants/permissions');
 // validate two dapps instead of 3
 const IS_FIREFOX = process.env.SELENIUM_BROWSER === Browser.FIREFOX;
 
-async function openDappAndSwitchChain(driver, dappUrl, chainId) {
+type ExpectedDetails = {
+  chainId: string;
+  networkText: string;
+  originText: string;
+};
+
+async function openDappAndSwitchChain(
+  driver: Driver,
+  dappUrl: string,
+  chainId?: string,
+): Promise<void> {
   // Open the dapp
-  await openDapp(driver, undefined, dappUrl);
+  const testDapp = new TestDapp(driver);
+  await testDapp.openTestDappPage({ url: dappUrl });
 
   // Connect to the dapp
-  await driver.clickElement({ text: 'Connect', tag: 'button' });
+  await testDapp.clickConnectAccountButton();
 
   await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
 
-  await driver.clickElementAndWaitForWindowToClose({
-    text: 'Connect',
-    tag: 'button',
-  });
+  await testDapp.clickConnectAccountButtonAndWaitForWindowToClose();
```

Comment:
mmm I think we should get rid of this method, as it doesn't use the "Test Dapp" selectors, rather the Dialog MetaMask selectors. That's an incorrect pattern, and we should eventually get rid of this mthod altogether.

Instead we should do this:

```
await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
const connectAccountConfirmation2 = new ConnectAccountConfirmation(
            driver,
          );
await connectAccountConfirmation2.checkPageIsLoaded();
await connectAccountConfirmation2.confirmConnect();
```

There are a couple of places where this method is being used. I suggest either you also fix those, or if you want to leave it out of scope of the PR, then here we don't introduce it here